dashboard/app/models/ability.rb

```diff
@@ -103,7 +103,7 @@ def initialize(user)
 
         if Pd::CourseFacilitator.exists?(facilitator: user, course: Pd::Workshop::COURSE_CSF)
           can :create, Pd::Workshop
-          can :update, Pd::Workshop, facilitators: {id: user.id}
+          can [:update, :destroy], Pd::Workshop, facilitators: {id: user.id}
```

Comment:
Then on the server, this also looks like _if_ the user is a CSF facilitator, then we are giving them permission to destroy any of their own workshops (CSF or not).  Maybe this is a non-issue in practice?

Reply:
Your explanation matches my understanding of how this works - I think in practice this is mostly fine, but I can think of one potential edge case:

If the user was both a CSF facilitator and also a CSD or CSP facilitator - then I think they would be able to delete their CSD and CSP workshops, which shouldn't be allowed. They could also create CSD or CSP workshops, which we don't want to allow. I think we could get around this by enforcing the course of the workshops they can create and delete, but I don't know if this is a concern practically speaking.

Reply:
Also - do program managers ever create CSF workshops for facilitators? should the facilitators be allowed to delete workshops that they didn't create ("organize")?

Reply:
So if the user is a CSF facilitator, they will be shown the client-side button to delete any workshop, but it will only end up working for their own workshops? Checked with ed team and it sounds like CSF facilitators facilitating non-CSF workshops happens enough that this is worth fixing to both only show the delete option for CSF, and restrict deletion to CSF

Reply:
To @clareconstantine's addition: no, CSF facilitators should only be able to delete workshops they created. so long as we're able to track this, they should not have the ability to delete workshops created for them by a program manager

Reply:
After some discussion, we made two changes: a CSF facilitator can still create a workshop, but now only a CSF one; and a CSF facilitator can now destroy a workshop, but only if they are the organizer of it.